eyelid-match-ops/src/primitives/poly/modular_poly/mul.rs

```diff
@@ -19,6 +19,7 @@ impl<C: PolyConf> MulAssign<C::Coeff> for Poly<C> {
         for coeff in &mut self.0.coeffs {
             *coeff *= rhs;
         }
+        self.truncate_to_canonical_form();
```

Comment:
This fix means we don't need to repeat these checks in the yashe code. (Unless we're working with raw coefficients.)